lib/ui/survey_question/answer/answer_form.dart

```diff
@@ -12,13 +15,30 @@ class AnswerForm extends StatelessWidget {
   });
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final controllers =
```

Comment:
@Wadeewee we need to dispose every controllers too

Reply:
@manh-t, What do you think if I call `.dispose` after `.clear` at line `31` instead of `override dispose`, if that I need to extend `ConsumerStatefulWidget` instead of `ConsumerWidget` 🤔 Do you have any suggestion 🙏  cc @hoangnguyen92dn

Reply:
If I'm not wrong by using `ConsumerWidget` we don't need to `dispose` of the stream just calling `.clear` is enough 🤔 Please correct me if I missed something 🙇‍♀️

Reply:
FYI: I fixed it by disposing of the `_pageController` in the parent view in  `SurveyQuestionsScreen`

Reply:
> Any issue if we update to use the ConsumerStatefulWidget

@hoangnguyen92dn, Actually, I wanted to avoid creating a separate state for the `sub-widget class`. However, I needed to access the current data when the `next button` is `triggered` from the `view model`. 

> since the form contains textfield to input the information, we should use a stateful widget instead.

To achieve this, I used the `ConsumerWidget` class, which is a `StatelessWidget version` of `ConsumerStatefulWidget`. Since we have already defined `ConsumerStatefulWidget` in the `parent class`, there is no need to use it again in the `child widget`.

This widget class is responsible for controlling the content within `PageView.builder` using `_pageController` defined in the `main class(SurveyQuestionsScreen)`. When we `dispose` of `_pageController`, all the instances associated including the `TextEditingController` with this class will be automatically cleaned up.

That all was I thought, Please correct me If I missed something 🙇‍♀️ cc @manh-t

Reply:
@Wadeewee sorry for late reply.

> What do you think if I call .dispose after .clear at line 31 instead of override dispose

The `dispose` should only be called when the widget is destroyed, not when we do any action like pressing a button. This is not a good practice 🙏 

> If I'm not wrong by using ConsumerWidget we don't need to dispose of the stream just calling .clear is enough 🤔 Please correct me if I missed something 🙇‍♀️

`clear` function is to clear the text in the textfield, it is totally different from `dispose`.

The best practice is when we define a controller in any widget, we should convert that widget to `Stateful` (or `ConsumerStatefulWidget` in riverpod) and add the `dispose` for that controller 🙏 

> When we dispose of _pageController, all the instances associated including the TextEditingController with this class will be automatically cleaned up.

Are you sure about this? When you destroy a widget without calling `dispose` of its controller. The controller is still there, we can't see it but it still takes some memory. Let's debug and call it after you destroy the widget and you will see 😄

Reply:
> What do you think if I call .dispose after .clear at line 31 instead of override dispose
> 
> The dispose should only be called when the widget is destroyed, not when we do any action like pressing a button. This is not a good practice 🙏

Agreed @manh-t for this that's why I said on this comment here 👇 
https://github.com/Wadeewee/survey-flutter-ic-pooh/pull/66#discussion_r1223806588

And regarding this 👇  let me try to debug as your suggestion if the `controller` is still there I will use `ConsumerStatefulWidget` instead 🚀  cc @hoangnguyen92dn 

> >When we dispose of _pageController, all the instances associated including the TextEditingController with this class will be automatically cleaned up.
> 
> Are you sure about this? When you destroy a widget without calling dispose of its controller. The controller is still there, we can't see it but it still takes some memory. Let's debug and call it after you destroy the widget and you will see

Reply:
@manh-t @hoangnguyen92dn, I think we don't need to use `TextEditingController` for the current implementation. Therefore, we can eliminate creating a separate `state` for the `sub-widget class`. I didn't encounter any issues when I removed it and tested it. Please see more in this [commit](https://github.com/Wadeewee/survey-flutter-ic-pooh/pull/66/commits/b15ad39df7ff525e4265d872216963008b3c43c7) and here're some POW and log 🙏 

https://github.com/Wadeewee/survey-flutter-ic-pooh/assets/28002315/7589cd13-9891-43c1-9474-17e5024dbb13


https://github.com/Wadeewee/survey-flutter-ic-pooh/assets/28002315/680e2ce5-7dcf-47b5-956b-e25bd36d65be

```
flutter:
flutter: unhandled element filter; Picture key: Svg loader
flutter: SubmitSurveyAnswerModel(037574cb93d16800eecd, 3)
flutter: SubmitSurveyAnswerModel(d43e7db9dedf66326795, 3)
flutter: SubmitSurveyAnswerModel(d43e7db9dedf66326795, 3)
flutter: SubmitSurveyAnswerModel(534f04bca9d051c8deb3, 3)
flutter: SubmitSurveyAnswerModel(d3b786f5c0b9eca4a863, 3)
flutter: SubmitSurveyAnswerModel(f688cf66f84988a495aa, 3)
flutter: SubmitSurveyAnswerModel(b9e6cf7a06deb810bbc7, 3)
flutter: SubmitSurveyAnswerModel(2a49e148c5b170aca804, g)
flutter: SubmitSurveyAnswerModel(491d49dd6b8174456acf, g)
flutter: SubmitSurveyAnswerModel(6db6dcae1a6c6644d723, 5)
flutter: SubmitSurveyAnswerModel(575db8c074601994bde3, b@n)
```

Reply:
@Wadeewee thanks for the dedicated effort to check it. Anw, I'm fine with this approach too 😁 Resolved ✅